### PR TITLE
feat: form instance support getFieldInstance

### LINKED ISSUE #23659: `form.getFieldInstance()` like v3
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
```js
form.getFieldInstance(...).focus()
```

### What does the proposed API look like?
```js
const [form] = Form.useForm();

form.getFieldInstance(['abc','efg']).focus()
```

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

resolve #23659

### 💡 Background and solution

v3 里内部做 scroll 用，在文档没有透出。 #23659 里看不少人想要，做成正式 API 给用户用好了。

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Form instance support `getFieldInstance`.       |
| 🇨🇳 Chinese |     Form 实例支持 `getFieldInstance`。    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/form/demo/ref-item.md](https://github.com/ant-design/ant-design/blob/form-instance/components/form/demo/ref-item.md)
[View rendered components/form/index.en-US.md](https://github.com/ant-design/ant-design/blob/form-instance/components/form/index.en-US.md)
[View rendered components/form/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/form-instance/components/form/index.zh-CN.md)
[View rendered components/progress/demo/segment.md](https://github.com/ant-design/ant-design/blob/form-instance/components/progress/demo/segment.md)
[View rendered components/progress/index.en-US.md](https://github.com/ant-design/ant-design/blob/form-instance/components/progress/index.en-US.md)
[View rendered components/progress/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/form-instance/components/progress/index.zh-CN.md)

### CODE REVIEW

**Comment by zombieJ** [comment]
> v3 里内部做 scroll 用，在文档没有透出。 #23659 里看不少人想要，做成正式 API 给用户用好了。

帮忙看看是不是有必要

File: components/form/demo/ref-item.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+```jsx
+import React from 'react';
+import { Button, Form, Input } from 'antd';
+
+const Demo = () => {
+  const [form] = Form.useForm();
+  const ref = React.useRef();
+
+  return (
+    <Form form={form} initialValues={{ list: ['light'] }} itemReferable>
```
Comment (afc163) [inline components/form/demo/ref-item.md:27]: 默认为 true 呢。

Comment (zombieJ) [reply components/form/demo/ref-item.md:27]: 默认为 true 就会有 v3 自定义 FC 组件报错不能用 ref 的问题。

Comment (zombieJ) [reply components/form/demo/ref-item.md:27]: v4 里用 id 查找元素代替过去 ref 后 `findDOMNode` 做了解耦，所以 v4 里不需要额外的 ref。

File: components/form/Form.tsx [hunk 2]
```diff
@@ -25,27 +24,32 @@ export interface FormProps extends Omit<RcFormProps, 'form'> {
   form?: FormInstance;
   size?: SizeType;
   scrollToFirstError?: boolean;
+  itemReferable?: boolean;
```
Comment (afc163) [inline components/form/Form.tsx:27]: itemReferable 和 getFieldInstance 太难联系到一起了，看上去毫无关系。

**Review by afc163 — APPROVED** [review]

**Comment by yakkantiraja** [comment]
I have taken latest upgrade but this fix not available. When this fix will be available to use ?

**Pull request merged**